pkg/kubectl/run.go

```diff
@@ -282,6 +302,94 @@ func (JobV1Beta1) Generate(genericParams map[string]interface{}) (runtime.Object
 	return &job, nil
 }
 
+type JobV1 struct{}
+
+func (JobV1) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"labels", false},
+		{"default-name", false},
+		{"name", true},
+		{"image", true},
+		{"port", false},
+		{"hostport", false},
+		{"stdin", false},
+		{"leave-stdin-open", false},
+		{"tty", false},
+		{"command", false},
+		{"args", false},
+		{"env", false},
+		{"requests", false},
+		{"limits", false},
+		{"restart", false},
+	}
+}
+
+func (JobV1) Generate(genericParams map[string]interface{}) (runtime.Object, error) {
+	args, err := getArgs(genericParams)
+	if err != nil {
+		return nil, err
+	}
+
+	envs, err := getV1Envs(genericParams)
```

Comment:
I was hoping to avoid the code duplication. The only alternative I could think of is to create an internal extensions.Job, as in v1beta1, and then explicitly convert it to batch.Job.